src/util/git-ref-resolver.js

```diff
@@ -0,0 +1,192 @@
+/* @flow */
+
+import type Config from '../config.js';
+import {removeSuffix} from './misc.js';
+
+const semver = require('semver');
+
+export type GitRefs = {
+  [name: string]: string,
+};
+export type DefaultBranch = {defaultBranch: true};
+export type ResolvedSha = {sha: string, ref: ?string};
+type Names = {tags: Array<string>, branches: Array<string>};
+
+export const isCommitSha = (target: string): boolean => Boolean(target) && /^[a-f0-9]{5,40}$/.test(target);
```

Comment:
I think it was intentional because there is a test for that specific case. And after thinking about it, I think it may help keeping the sha consistently lowercase over all the application (except user input)